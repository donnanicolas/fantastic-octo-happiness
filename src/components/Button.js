// @flow
import { omit } from 'lodash';
import React, { PureComponent } from 'react';
import { Button as BSButton } from 'react-bootstrap';
import styled from 'styled-components';

import { green } from '../utils/spotify';

const omittedProps = ['primary', 'success', 'info', 'warning', 'danger', 'link', 'sm', 'lg', 'xs'];

type Props = {
  primary?: boolean,
  success?: boolean,
  info?: boolean,
  warning?: boolean,
  danger?: boolean,
  link?: boolean,
  spotify?: boolean,
  lg?: boolean,
  sm?: boolean,
  xs?: boolean,
  glyph?: boolean,
  [x: string]: any,
};

const StyledButton = styled(props => <BSButton {...omit(props, ['spotify', 'glyph']) } />) `
  background-color: ${({ spotify }: Props) => spotify ? green : ''};
  color: ${({ spotify }: Props) => spotify ? '#000' : ''};
  font-family: 'Montserrat', sans-serif;


  &.btn-primary {
    border: none;
    background-color: #000;
    &:hover {
      border: none;
    }
  }

  &.btn-danger {
    background-color: #DB4C2C;
    border: none;
    &:hover {
      border: none;
    }
  }

  ${({ glyph }: Props) => glyph ? (`
    .glyphicon {
      float: right;
      margin-top: 1px;
    }
  `) : (
    ''
  )}
`;

const bsStyle = ({ primary, success, info, warning, danger, link }: Props) => {
  if (primary) return 'primary';
  if (success) return 'success';
  if (info) return 'info';
  if (warning) return 'warning';
  if (danger) return 'danger';
  if (link) return 'link';
  return null;
}

const bsSize = ({ lg, sm, xs }: Props) => {
  if (lg) return 'large';
  if (sm) return 'small';
  if (xs) return 'xsmall';
  return null;
}

class Button extends PureComponent {
  props: Props;

  render() {
    const props = this.props;
    const others = omit(this.props, omittedProps);
    return <StyledButton bsStyle={bsStyle(props)} bsSize={bsSize(props)} {...others} />;
  }
}

export default Button;
