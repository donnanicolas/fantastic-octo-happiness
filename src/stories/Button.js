// @flow
import React from 'react';
import { Glyphicon } from 'react-bootstrap';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import Button from '../components/Button';

const addStories = () => {
  storiesOf('Button', module)
    .add('default', () => <Button onClick={action('clicked')}>default</Button>)
    .add('primary', () => <Button primary onClick={action('clicked')}>primary</Button>)
    .add('success', () => <Button success onClick={action('clicked')}>success</Button>)
    .add('info', () => <Button info onClick={action('clicked')}>info</Button>)
    .add('warning', () => <Button warning onClick={action('clicked')}>warning</Button>)
    .add('danger', () => <Button danger onClick={action('clicked')}>danger</Button>)
    .add('link', () => <Button link onClick={action('clicked')}>link</Button>)
    .add('spotify', () => <Button spotify onClick={action('clicked')}>Spotify</Button>)
    .add('with icon', () =>
      <Button glyph primary onClick={action('clicked')}>
        Play
        <Glyphicon glyph="play" />
      </Button>
    );

};

export default addStories;