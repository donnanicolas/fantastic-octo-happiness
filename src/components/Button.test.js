import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

it('renders a basic button', () => {
    const tree = renderer.create(
    <Button>Basic Button</Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a primary button', () => {
    const tree = renderer.create(
    <Button primary>Basic Button</Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a success button', () => {
    const tree = renderer.create(
    <Button success>Basic Button</Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a info button', () => {
    const tree = renderer.create(
    <Button info>Basic Button</Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a warning button', () => {
    const tree = renderer.create(
    <Button warning>Basic Button</Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a danger button', () => {
    const tree = renderer.create(
    <Button danger>Basic Button</Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders a link button', () => {
    const tree = renderer.create(
    <Button link>Basic Button</Button>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
