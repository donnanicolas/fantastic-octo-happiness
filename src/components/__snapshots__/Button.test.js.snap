// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a basic button 1`] = `
<button
  className="btn"
  disabled={false}
  type="button"
>
  Basic Button
</button>
`;

exports[`renders a danger button 1`] = `
<button
  className="btn btn-danger"
  disabled={false}
  type="button"
>
  Basic Button
</button>
`;

exports[`renders a info button 1`] = `
<button
  className="btn btn-info"
  disabled={false}
  type="button"
>
  Basic Button
</button>
`;

exports[`renders a link button 1`] = `
<button
  className="btn btn-link"
  disabled={false}
  type="button"
>
  Basic Button
</button>
`;

exports[`renders a primary button 1`] = `
<button
  className="btn btn-primary"
  disabled={false}
  type="button"
>
  Basic Button
</button>
`;

exports[`renders a success button 1`] = `
<button
  className="btn btn-success"
  disabled={false}
  type="button"
>
  Basic Button
</button>
`;

exports[`renders a warning button 1`] = `
<button
  className="btn btn-warning"
  disabled={false}
  type="button"
>
  Basic Button
</button>
`;
